### PR TITLE
add test to check crosstab supports Float64 formats

### LINKED ISSUE #50313: QST: Pandas Crosstab dos not support Float (with capital F) number formats
### Research

- [X] I have searched the [[pandas] tag](https://stackoverflow.com/questions/tagged/pandas) on StackOverflow for similar questions.

- [X] I have asked my usage related question on [StackOverflow](https://stackoverflow.com).

### Link to question on StackOverflow

https://stackoverflow.com/questions/74830329/pandas-crosstab-dos-not-support-float-with-capital-f-number-formats

### Question about pandas

Hello guys,
I´ve noted that pandas crosstab does not support Float formats (instead of float). What I really would like is to understand why and if the output error message could be clearer to understand?
Best regards.

I am working on a sample data transaction dataframe. Such base contains cliente ID, transaction gross value (GMV) and revenue. Take this example as DF :

```
df = pd.DataFrame({
    'id' :  [1,2,3,4,5],
    'date' : [datetime(2022,6,1),datetime(2022,12,31),datetime(2022,12,31),datetime(2022,12,31),datetime(2022,12,31)],
    'revenue' : [10.1, 10.1, 10.1, 10.1, 10.1]})
df = df.astype({'revenue':'Float64'})
```

Now I create a crosstab

`CrossTab = pd.crosstab(df['id'], df['date'], values=df['revenue'], rownames=None, colnames=None, aggfunc='sum', margins=True, margins_name='All', dropna=False, normalize=False)`

the output

```
Output exceeds the [size limit](command:workbench.action.openSettings?[). Open the full output data [in a text editor](command:workbench.action.openLargeOutput?62540584-615b-4486-a364-032ed486e331)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[3], line 1
----> 1 CrossTab = pd.crosstab(df['id'], df['date'], values=df['revenue'], rownames=None, colnames=None, aggfunc='sum', margins=True, margins_name='All', dropna=False, normalize=False)

File c:\Users\fabio\AppData\Local\Programs\Python\Python311\Lib\site-packages\pandas\core\reshape\pivot.py:691, in crosstab(index, columns, values, rownames, colnames, aggfunc, margins, margins_name, dropna, normalize)
    688     df["__dummy__"] = values
    689     kwargs = {"aggfunc": aggfunc}
--> 691 table = df.pivot_table(
    692     "__dummy__",
    693     index=unique_rownames,
    694     columns=unique_colnames,
    695     margins=margins,
    696     margins_name=margins_name,
    697     dropna=dropna,
    698     **kwargs,
    699 )
    701 # Post-process
    702 if normalize is not False:

File c:\Users\fabio\AppData\Local\Programs\Python\Python311\Lib\site-packages\pandas\core\frame.py:8728, in DataFrame.pivot_table(self, values, index, columns, aggfunc, fill_value, margins, dropna, margins_name, observed, sort)
   8711 @Substitution("")
   8712 @Appender(_shared_docs["pivot_table"])
   8713 def pivot_table(
   (...)
...
--> 292     raise TypeError(dtype)  # pragma: no cover
    294 converted = maybe_downcast_numeric(result, dtype, do_round)
    295 if converted is not result:

TypeError: Float64
```

### PR DESCRIPTION
**Pull request by natmokval**
- [x] closes #50313
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

A  new test is added. The test validates `crosstab` behavior with `Float64`. I have checked that the problem occurs when `Float64` is used together with `margins=True`. 
The test fails on `1.5.2` and passes on `main`.

### CODE REVIEW

File: pandas/tests/reshape/test_crosstab.py [hunk 1]
```diff
@@ -795,6 +795,33 @@ def test_margin_normalize_multiple_columns(self):
         expected.index.name = "C"
         tm.assert_frame_equal(result, expected)
 
+    def test_margin_support_Float(self):
+        # GH 50313
+        # use Float64 formats and function aggfunc with margins
+        df = DataFrame(
+            {"A": [1, 2, 2, 1], "B": [3, 3, 4, 5], "C": [-1.0, 10.0, 1.0, 10.0]}
+        )
+        df = df.astype({"C": "Float64"})
```
Comment (phofl) [inline pandas/tests/reshape/test_crosstab.py:804]: Can you avoid the astype and define this as float immediately?

File: pandas/tests/reshape/test_crosstab.py [hunk 2]
```diff
@@ -795,6 +795,33 @@ def test_margin_normalize_multiple_columns(self):
...
+        expected = DataFrame(
+            [
+                [-1.0, pd.NA, 10.0, 9.0],
+                [10.0, 1.0, pd.NA, 11.0],
+                [9.0, 1.0, 10.0, 20.0],
+            ],
+            index=[1, 2, "All"],
+            dtype="Float64",
+        )
+        expected.columns = Index([3, 4, 5, "All"], dtype="object", name="B")
```
Comment (phofl) [inline pandas/tests/reshape/test_crosstab.py:821]: Same here, please define expected in one go

Comment (natmokval) [reply pandas/tests/reshape/test_crosstab.py]: Thank you, @phofl, for your comments. I updated my PR as you suggested.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @natmokval

**Comment by natmokval** [comment]
Thank you, @phofl, for reviewing my PR.

**Pull request merged**